lisa/trace.py

```diff
@@ -1142,29 +2248,36 @@ def _sanitize_CpuFrequency(self):
 
                 df.sort_index(inplace=True)
 
-            self._ftrace.cpu_frequency.data_frame = df
+            return df
 
-    def _sanitize_funcgraph(self):
+    def _sanitize_funcgraph(self, event, df, aspects):
         """
         Resolve the kernel function names.
         """
 
-        events = [
-            event
-            for event in ('funcgraph_entry', 'funcgraph_exit')
-            if self.has_events(event)
-        ]
-        if not events:
-            return
-
         try:
             addr_map = self.plat_info['kernel']['symbols-address']
         except KeyError as e:
             self.get_logger().warning('Missing symbol addresses, function names will not be resolved: {}'.format(e))
+            return df
         else:
-            for event in events:
-                df = self.df_events(event)
-                df['func_name'] = df['func'].map(addr_map)
+            df['func_name'] = df['func'].map(addr_map)
+            return df
+
+    _SANITIZATION_FUNCTIONS = {
```

Comment:
Could that be built from decorators? It would be a tad easier to maintain that way.

Reply:
That's what I wanted to do. Unfortunately, I could not find any way of writing such decorator in a way that could be used outside of that class as well.

EDIT: Just found a way. 
What does not work because of scoping issues (even with nonlocal, since it only seems to work with function nesting):
```
class C:
    l = []
    def decorator(f):
        l.append(f)
        return f

    @decorator
    def myfunc():
        pass
```

What does work:
```
class C:
    l = []
    def decorator(f, l=l):
        nonlocal l
        l.append(f)
        return f

    @decorator
    def myfunc():
        pass


print(C.l)
```